app/models/zone.rb

```diff
@@ -148,6 +148,10 @@ def ems_middlewares
     ext_management_systems.select { |e| e.kind_of? ManageIQ::Providers::MiddlewareManager }
   end
 
+  def middleware_servers
+    ems_middlewares.flat_map(&:middleware_servers)
```

Comment:
I think it is not needed on this case as it is invoking ems_middleware call to get the association.